src/modules/hip/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,1433 @@
+    {
+        if (hipThreadIdx_x < threadMax)
+            partialSumRowPtr_smem[hipThreadIdx_x] += partialSumRowPtr_smem[hipThreadIdx_x + threadMax];
+        __syncthreads();
+    }
+
+    if (hipThreadIdx_x == 0)
+    {
+        // Reduction of 16 floats on 16 threads per block in y dimension
+        for (int threadMax = 8, increment = 128; threadMax >= 1; threadMax /= 2, increment /= 2)
+        {
+            if (hipThreadIdx_y < threadMax)
+                partialSumRowPtr_smem[0] += partialSumRowPtr_smem[increment];
+            __syncthreads();
+        }
+
+        // Final store to dst
+        if (hipThreadIdx_y == 0)
+            tensorSumArr[(hipBlockIdx_z * hipGridDim_y + hipBlockIdx_y) * hipGridDim_x + hipBlockIdx_x] = partialSumRowPtr_smem[0];
+    }
+}
+
+// Handle U8
+__global__ void tensor_sum_pln3(Rpp8u *srcPtr,
+                                uint3 srcStridesNCH,
+                                Rpp32u *tensorSumArr,
+                                RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    __shared__ uint partialRSum_smem[16][16];                                                   // 16 rows of src, 128 reduced cols of src in a 16 x 16 thread block
+    __shared__ uint partialGSum_smem[16][16];
+    __shared__ uint partialBSum_smem[16][16];
+    uint *partialRSumRowPtr_smem = &partialRSum_smem[hipThreadIdx_y][0];                        // uint pointer to beginning of each row in Shared
+    uint *partialGSumRowPtr_smem = &partialGSum_smem[hipThreadIdx_y][0];
+    uint *partialBSumRowPtr_smem = &partialBSum_smem[hipThreadIdx_y][0];
+    partialRSumRowPtr_smem[hipThreadIdx_x] = 0;                                                 // initialization of Shared to 0 using all 16 x 16 threads
+    partialGSumRowPtr_smem[hipThreadIdx_x] = 0;
+    partialBSumRowPtr_smem[hipThreadIdx_x] = 0;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    int xAlignedLength = roiTensorPtrSrc[id_z].xywhROI.roiWidth & ~7;                           // alignedLength for vectorized global loads
+    int xDiff = roiTensorPtrSrc[id_z].xywhROI.roiWidth - xAlignedLength;                        // difference between roiWidth and alignedLength
+    uint srcIdx = (id_z * srcStridesNCH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+
+    d_uint24 src_ui24;
+    rpp_hip_load24_pln3_to_uint24_pln3(srcPtr + srcIdx, srcStridesNCH.y, &src_ui24);
+
+    if (id_x + 8 > roiTensorPtrSrc[id_z].xywhROI.roiWidth)                                      // local memory reset of invalid values (from the vectorized global load) to 0
+    {
+        for(int i = xDiff; i < 8; i++)
```

Comment:
we need to revisit this code segment. I am not sure if this is the right way to prevent invalid load of pixels. Also have a loop here would degrade the overall performance of this kernel